docs/fastdds/dynamic_types/supported_types.rst

```diff
@@ -0,0 +1,311 @@
+modifies the involved bits instead of the full primitive value.
+
+Bit_bound and position of the bitfield can be set using annotations (useful when converting between static and dynamic
+bitsets).
+
+.. literalinclude:: /../code/CodeTester.cpp
+   :language: c++
+   :start-after: //DYNAMIC_TYPES_CREATE_BITSETS
+   :end-before: //!--
+   :dedent: 8
+
+Bitsets allows inheritance, exactly with the same OOP meaning. To inherit from another bitset, we must create the
+bitset calling the ``create_child_struct_builder`` of the factory. This function is shared with structures and will
+deduce our type depending on the parent's type.
+
+.. literalinclude:: /../code/CodeTester.cpp
+   :language: c++
+   :start-after: //DYNAMIC_TYPES_CREATE_BITSETS-INHERIT
+   :end-before: //!--
+   :dedent: 8
+
+.. _dynamictypes_supportedtypes_union:
+
+Union
+-----
+
+Unions are a special kind of structures where only one of the members is active
+at the same time.
+To control these members, users must set the ``discriminator`` type that is going to be used
+to select the current member calling the ``create_union_builder`` function.
+The ``discriminator`` itself is a Dynamic Type of any primitive type, string type or union type.
+
+Every member that is going to be added needs at least one ``union_case_index`` to set
+how it is going to be selected and, optionally, if it is the default value of the union.
+
+.. literalinclude:: /../code/CodeTester.cpp
+   :language: c++
+   :start-after: //DYNAMIC_TYPES_CREATE_UNIONS
+   :end-before: //!--
+   :dedent: 8
+
+.. _dynamictypes_supportedtypes_sequence:
+
+Sequence
+--------
+
+A complex type that manages its members as a list of items allowing users to
+insert, remove or access to a member of the list. To create this type users
+need to specify the type that it is going to store and optionally the size
+limit of the list.
+
+To ease the memory management of this type, ``DynamicData`` has these functions:
+
+ - ``insert_sequence_data``: Creates a new element at the end of the list and returns
+   the ``id`` of the new element.
+ - ``remove_sequence_data``: Removes the element of the given index and refreshes the ids
+   to keep the consistency of the list.
+ - ``clear_data``: Removes all the elements of the list.
+
+.. literalinclude:: /../code/CodeTester.cpp
+   :language: c++
+   :start-after: //DYNAMIC_TYPES_CREATE_SEQUENCES
+   :end-before: //!--
+   :dedent: 8
+
+.. _dynamictypes_supportedtypes_array:
+
+Array
+-----
+
+Arrays are pretty similar to sequences with two main differences:
+they can have multiple dimensions and they don't need that the elements
+are stored consecutively.
+
+An array needs to know the number of dimensions it is managing.
+For that, users must provide a vector with as many elements as dimensions in the array.
+Each element in the vector represents the size of the given dimension.
+If the value of an element is set to zero, the default value applies ( :class:`100` ).
```

Comment:
A compilation error:
create_array_builder takes a `std::vector<uint32_t>`. If you try with a `std::vector<int32_t>`, it fails, even if the values are all positive.
If you try to set a negative on a ''std::vector<uint32_t>'' it will also fail